server/shadow/shadow_encoder.c

```diff
@@ -130,9 +130,6 @@ int shadow_encoder_uninit_grid(rdpShadowEncoder* encoder)
 
 int shadow_encoder_init_rfx(rdpShadowEncoder* encoder)
 {
-	rdpContext* context = (rdpContext*) encoder->client;
-	rdpSettings* settings = context->settings;
-
 	if (!encoder->rfx)
```

Comment:
I may misunderstand your question. The rfx_context_new shows if it fail, it always returns NULL.
Are you actually asking "How encoder->rfx is not NULL"? The answer is that it might be called multiple times with 'shadow_encoder_prepare'. So if we've already created context, we don't create again.